### PR TITLE
Refactors part 3: Refactor Resource to Attribute

### PR DESCRIPTION
**Pull request by fppt**

### CODE REVIEW

File: grakn-graph/src/main/java/ai/grakn/graph/internal/concept/TypeImpl.java [hunk 1]
```diff
@@ -390,33 +390,33 @@ public T property(Schema.VertexProperty key, Object value){
...
     /**
      * Creates a relation type which allows this type and a resource type to be linked.
-     * @param resourceType The resource type which instances of this type should be allowed to play.
+     * @param attributeType The resource type which instances of this type should be allowed to play.
      * @param has the implicit relation type to build
      * @param hasValue the implicit role type to build for the resource type
      * @param hasOwner the implicit role type to build for the type
      * @param required Indicates if the resource is required on the entity
      * @return The Type itself
      */
```
Comment (aelred) [inline grakn-graph/src/main/java/ai/grakn/graph/internal/concept/TypeImpl.java]: s/resource/attribute in a few places here.

File: docs/pages/documentation/3-querying-with-graql/compute-queries.md [hunk 2]
```diff
@@ -52,12 +52,12 @@ to you. A summary of the statistics algorithms is given in the table below.
...
-| [`median`](#mean)    | Compute the median value of a resource.                           |
-| [`min`](#minimum)    | Compute the minimum value of a resource. |
-| [`std`](#standard-deviation)    | Compute the standard deviation of a resource. |
-| [`sum`](#sum)    | Compute the sum of a resource. |
+| [`max`](#maximum)    | Compute the maximum value of a attribute. |
+| [`mean`](#mean)    | Compute the mean value of a attribute.                           |
+| [`median`](#mean)    | Compute the median value of a attribute.                           |
+| [`min`](#minimum)    | Compute the minimum value of a attribute. |
+| [`std`](#standard-deviation)    | Compute the standard deviation of a attribute. |
+| [`sum`](#sum)    | Compute the sum of a attribute. |
```
Comment (fppt) [inline docs/pages/documentation/3-querying-with-graql/compute-queries.md]: hahahahaha

**Comment by fppt** [comment]
You review has shown me that we should always refactor to words not starting with vowels

**Comment by fppt** [comment]
@aelred Reive por favor

File: grakn-graph/src/main/java/ai/grakn/graph/internal/concept/TypeImpl.java [hunk 3]
```diff
@@ -389,27 +389,27 @@ public T property(Schema.VertexProperty key, Object value){
     }
 
     /**
-     * Creates a relation type which allows this type and a resource type to be linked.
-     * @param attributeType The resource type which instances of this type should be allowed to play.
+     * Creates a relation type which allows this type and a {@link ai.grakn.concept.Attribute} type to be linked.
```
Comment (aelred) [inline grakn-graph/src/main/java/ai/grakn/graph/internal/concept/TypeImpl.java]: You're not done yet! ✨  (sorry)

**Comment by fppt** [comment]
!rtg

**Pull request merged**